src/integrationTest/java/io/aiven/kafka/connect/jdbc/oracle/VerifyInsertIT.java

```diff
@@ -0,0 +1,189 @@
+            for (int partition = 0; partition < TEST_TOPIC_PARTITIONS; partition++) {
+                final String recordName = "user-" + i;
+                final String recordValue = "value-" + i;
+                final ProducerRecord<String, GenericRecord> msg = createRecord(i, partition, recordName, recordValue);
+                sendFutures.add(producer.send(msg));
+            }
+        }
+        producer.flush();
+        for (final Future<RecordMetadata> sendFuture : sendFutures) {
+            sendFuture.get();
+        }
+    }
+
+    @AfterEach
+    public void afterEach() throws SQLException {
+        executeSqlStatement(DROP_TABLE);
+    }
+
+    @Test
+    public void testSinkConnector() throws Exception {
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        executeSqlStatement(CREATE_TABLE);
+        consumer.assign(Collections.singleton(new TopicPartition(TEST_TOPIC_NAME, 0)));
+        // Start the sink connector
+        connectRunner.createConnector(basicSinkConnectorConfig());
+
+        // Send test data to Kafka topic
+        sendTestData(1);
+
+        await().atMost(Duration.ofSeconds(20)).pollInterval(Duration.ofSeconds(19))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(1);
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).column("ID")
+                            .value().isEqualTo("0");
+                });
+    }
+
+    @Test
+    public void testSinkWithPKModeRecordKeyConnector() throws Exception {
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        executeSqlStatement(CREATE_TABLE_WITH_PK);
+        consumer.assign(Collections.singleton(new TopicPartition(TEST_TOPIC_NAME, 0)));
+        // Start the sink connector
+        connectRunner.createConnector(sinkConnectorConfigWithPKModeRecordKey());
+
+        // Send test data to Kafka topic
+        sendTestData(1);
+
+        await().atMost(Duration.ofSeconds(20)).pollInterval(Duration.ofSeconds(19))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(1);
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).column("ID")
+                            .value().isEqualTo("0");
+                });
+    }
+
+    @Test
+    public void testSinkConnectorInDeleteMode() throws Exception {
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        executeSqlStatement(CREATE_TABLE);
+        consumer.assign(Collections.singleton(new TopicPartition(TEST_TOPIC_NAME, 0)));
+        // Start the sink connector
+        final Map<String, String> config = basicSinkConnectorConfig();
+        config.put("pk.mode", "record_key");
+        config.put("pk.fields", "id"); // assigned name for the primitive key
```

Comment:
We can simplify this, right?
```suggestion
        final Map<String, String> config = sinkConnectorConfigWithPKModeRecordKey();
```